server/controllers/finance/transferts.js

```diff
@@ -0,0 +1,145 @@
+
+    qLimit  = Number(qLimit);
+
+    sql += 'WHERE pc.origin_id = ? ORDER BY pc.date DESC ';
+    sql += (qLimit) ? 'LIMIT ' + Math.floor(qLimit) : '';
+
+    return db.exec(sql, [row[0].id]);
+  })
+  .then(function (rows) {
+    if (!rows.length) { next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+
+}
+
+/**
+* GET /transferts/:id
+*
+* @desc This function is responsible for getting a particular transfert record from the database
+*/
+function getDetail(req, res, next) {
+  var sql,
+      qUuid  = req.params.id;
+
+  /** Getting the transfert module id */
+  sql = 'SELECT pcm.id FROM primary_cash_module pcm WHERE pcm.text = "transfert" ';
+
+  db.exec(sql)
+  .then(function (row) {
+    /** Getting transfert records */
+    sql =
+        'SELECT pc.uuid, pc.project_id, pc.type, pc.date, pc.currency_id, pc.account_id, pc.cost, pc.description, ' +
+        'pc.cash_box_id, pc.origin_id, pc.user_id, pci.document_uuid ' +
+        'FROM primary_cash pc ' +
+        'JOIN primary_cash_item pci ON pci.primary_cash_uuid = pc.uuid ';
+
+    sql += 'WHERE pc.origin_id = ? AND pc.uuid = ? ORDER BY pc.date DESC ';
+
+    return db.exec(sql, [row[0].id, qUuid]);
+  })
+  .then(function (rows) {
+    if (!rows.length) { next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+}
+
+
+/**
+* POST /transferts
+*
+* @desc This function is responsible for creating a new transfert record in the database
+* and writing to the posting journal
+*/
+function create(req, res, next) {
+  var qData = req.body;
+
+
+  var missingRequiredParameters =
+    (
+      !qData.uuid && !qData.project_id && !qData.currency_id &&
+      !qData.account_id && !qData.cost && !qData.cash_box_id &&
+      !qData.origin_id && !qData.user_id
+    );
+  if (missingRequiredParameters) {
+    return next(new req.codes.ERR_MISSING_REQUIRED_PARAMETERS());
+  }
+
+  var sqlPrimaryCash = 'INSERT INTO primary_cash ' +
+      '(uuid, project_id, type, date, currency_id, account_id, cost, description, cash_box_id, origin_id, user_id) ' +
+      'VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)';
+  var paramPrimaryCash = [
+      qData.uuid, qData.project_id, qData.type, new Date(qData.date), qData.currency_id,
+      qData.account_id, qData.cost, qData.description, qData.cash_box_id, qData.origin_id, qData.user_id
+  ];
+
+  var sqlPrimaryCashItem = 'INSERT INTO primary_cash_item ' +
+      '(uuid, primary_cash_uuid, debit, document_uuid, credit) VALUES (?, ?, ?, ?, ?)';
+  var paramPrimaryCashItem = [uuid(), qData.uuid, qData.cost, qData.uuid, 0];
+
+  db.exec(sqlPrimaryCash, paramPrimaryCash)
+  .then(function (data) {
+    return db.exec(sqlPrimaryCashItem, paramPrimaryCashItem);
+  })
+  .then(function (data) {
+    journalTransfer(qData.uuid, qData.user_id, function (err, result) {
+      if (err) return next(err);
```

Comment:
This code is correct, and many projects use this syntax.  However, most of our code explicitly writes the brackets ...
```js
if (err) { return next(err); }
```
.. for clarity.  If we are going to begin omitting the brackets, we should decide that as a team.

@IMA-WorldHealth/local-contributors, what do you guys think?